src/client-agent/notify.c

```diff
@@ -39,42 +39,28 @@ char *getsharedfiles()
     return (ret);
 }
 
-#ifndef WIN32
 char *get_agent_ip()
 {
     char agent_ip[IPSIZE + 1] = { '\0' };
-#if defined (__linux__) || defined (__MACH__) || defined (sun) || defined(FreeBSD) || defined(OpenBSD)
-    int sock;
-    int i;
-    static const char * REQUEST = "host_ip";
-
-    for (i = SOCK_ATTEMPTS; i > 0; --i) {
-        if (sock = control_check_connection(), sock >= 0) {
-            if (OS_SendUnix(sock, REQUEST, strlen(REQUEST)) < 0) {
-                mdebug1("Error sending msg to control socket (%d) %s", errno, strerror(errno));
-            }
-            else{
-                if (OS_RecvUnix(sock, IPSIZE, agent_ip) <= 0) {
-                    mdebug1("Error receiving msg from control socket (%d) %s", errno, strerror(errno));
-                    agent_ip[0] = '\0';
-                }
-            }
-
-            close(sock);
-            break;
-        } else {
-            mdebug2("Control module not yet available. Remaining attempts: %d", i - 1);
-            sleep(1);
+#if defined (__linux__) || defined (__MACH__) || defined (sun) || defined(FreeBSD) || defined(OpenBSD) || defined(WIN32)
+    struct sockaddr_storage sas;
+    socklen_t len = sizeof(sas);
+    int err = getsockname(agt->sock, (struct sockaddr *)&sas, &len);
```

Comment:
As far as getstockname is concerned, the type of specialization, although correct, is unnecessary and even confusing if one takes into account that at the time `agt::sock` is assigned it is not specialized.

`connect_server(...) `
```c
    if (agt->server[server_id].protocol == IPPROTO_UDP) {
        agt->sock = OS_ConnectUDP(agt->server[server_id].port, ip_address, strchr(ip_address, ':') != NULL ? 1 : 0, agt->server[server_id].network_interface);
    } else {
        agt->sock = OS_ConnectTCP(agt->server[server_id].port, ip_address, strchr(ip_address, ':') != NULL ? 1 : 0, agt->server[server_id].network_interface);
    }
``` 
`OS_Connect(...)`
```c
    if (protocol == IPPROTO_TCP) {
        if ((ossock = socket(ipv6 == 1 ? AF_INET6 : AF_INET, SOCK_STREAM, IPPROTO_TCP)) < 0) {
            return (OS_SOCKTERR);
        }
    } else if (protocol == IPPROTO_UDP) {
        if ((ossock = socket(ipv6 == 1 ? AF_INET6 : AF_INET, SOCK_DGRAM, IPPROTO_UDP)) < 0) {
``` 
In any case, it would be more correct the changes to be global otherwise how this affects it would remain disconnected.

Reply:
In relation to the `getsockname` function and the correct operation even when the official documentation declares it out of range.
I think I have found the explanation about why it is working correctly in `Windows XP` in this snapshot:

<details>
<summary>you can see the same statement and data published in `MSDN (2004)`</summary>

![2023-05-22_16-31](https://github.com/wazuh/wazuh/assets/93778492/da2e53b9-4c2c-4005-aed5-1538a2d606c1)

</details>

This makes it clear that the `getsockname` function existed and was documented in versions prior to those currently released.

Reply:
> As far as getstockname is concerned, the type of specialization, although correct, is unnecessary and even confusing if one takes into account that at the time agt::sock is assigned it is not specialized.

Ok, you are right, thank you.

> This makes it clear that the getsockname function existed and was documented in versions prior to those currently released.

Thank you for the research and confirmation.
This explains why it works in my local tests.